clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -155,7 +155,16 @@ public void remove(String id) {
     public ConnectionState connectionState(String id) {
         return nodeState(id).state;
     }
-    
+
+    /**
+     * Check if we have attempted to connect to this node.
+     * @param id node id
+     * @return true if we have attempted to connect to the node
+     */
+    public boolean connectionAttempted(String id) {
```

Comment:
Good point. I think you suggestion is indeed more consistent with current methods in `ClusterConnectionState`. I have updated the patch accordingly.